sdk/storage/azure-storage-blob/azure/storage/blob/_shared/policies.py

```diff
@@ -620,3 +634,24 @@ def get_backoff_time(self, settings):
             if self.backoff > self.random_jitter_range else 0
         random_range_end = self.backoff + self.random_jitter_range
         return random_generator.uniform(random_range_start, random_range_end)
+
+
+class StorageBearerTokenCredentialPolicy(BearerTokenCredentialPolicy):
```

Comment:
for different clouds (mooncake, gov, etc) how are we accounting for this for the scopes?

Reply:
I'm not exactly sure how OAuth works in those clouds, but everything should just work. We are using the same default scope as we were previously `STORAGE_OAUTH_SCOPE` (it just moved into this file) and then, in the case of a challenge, we will follow whatever scope is provided in the challenge (from the service). I'm not sure if customers override `STORAGE_OAUTH_SCOPE` in certain clouds or what happens but it should continue to work, just now with added support for challenges.

Reply:
We should create an item to investigate and make sure that the Python SDK supports other clouds at some point.

Reply:
Sure. I've opened #23198 for that.